src/tabset/tabset.ts

```diff
@@ -83,10 +94,22 @@ export class NgbTabset implements AfterContentChecked {
    */
   @Input() type: string = 'tabs';
 
+  /**
+   * A tab change event fired right before the tab selection happens
+   */
+  @Output() change = new EventEmitter<NgbTabChangeEvent>();
+
   select(tabIdx: string) {
     let selectedTab = this._getTabById(tabIdx);
-    if (selectedTab && !selectedTab.disabled) {
-      this.activeId = selectedTab.id;
+    if (selectedTab && !selectedTab.disabled && this.activeId !== selectedTab.id) {
+      let preventSelection = false;
```

Comment:
No, no. Just rename `preventSelection` to `defaultPrevented`. The "public API" for users should still be `preventDefault()`